descartes_planner/test/planner_tests.cpp

```diff
@@ -0,0 +1,120 @@
+#include <gtest/gtest.h>
+
+#include <ros/console.h>
+
+#include <descartes_core/path_planner_base.h>
+#include <descartes_planner/dense_planner.h>
+#include <descartes_planner/sparse_planner.h>
+
+#include <descartes_trajectory_test/cartesian_robot.h>
+
+#include "utils/trajectory_maker.h"
+
+
+using namespace descartes_planner;
+using namespace descartes_core;
+
+// Factory methods for planner construction
+template <class T>
+PathPlannerBase* createPathPlanner();
+
+template <>
+PathPlannerBase* createPathPlanner<DensePlanner>()
+{
+  return new DensePlanner();
+}
+
+template <>
+PathPlannerBase* createPathPlanner<SparsePlanner>()
+{
+  return new SparsePlanner();
+}
+
+// Google Test Framework for Path Planners
+
+template <class T>
+class PathPlannerTest : public testing::Test
+{
+protected:
+  typedef T planner_type;
+
+  PathPlannerBasePtr makePlanner()
+  {
+    PathPlannerBasePtr planner = PathPlannerBasePtr(createPathPlanner<T>());
+    EXPECT_TRUE(planner->initialize(robot_));
+    return planner;
+  }
+
+  PathPlannerTest()
+    : velocity_limits_(6, 1.0)
+    , robot_(new descartes_trajectory_test::CartesianRobot(5.0, 0.001, velocity_limits_))
+  {}
+
+  std::vector<double> velocity_limits_;
+  RobotModelConstPtr robot_; 
+};
+
+using testing::Types;
+
+// Add types of trajectory points here:
+typedef Types<DensePlanner, SparsePlanner> Implementations;
```

Comment:
Since the planner is really an interface, is there a reason a [type-parameterized](https://github.com/google/googletest/blob/master/googletest/docs/AdvancedGuide.md#type-parameterized-tests) test case was not used?